utilities/memory_allocators.h

```diff
@@ -0,0 +1,65 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <atomic>
+
+#include "rocksdb/memory_allocator.h"
+
+namespace ROCKSDB_NAMESPACE {
+// A memory allocator using new/delete
+class DefaultMemoryAllocator : public MemoryAllocator {
+ public:
+  static const char* kClassName() { return "DefaultMemoryAllocator"; }
+  const char* Name() const override { return kClassName(); }
+  void* Allocate(size_t size) override {
+    return static_cast<void*>(new char[size]);
+  }
+
+  void Deallocate(void* p) override { delete[] static_cast<char*>(p); }
+};
+
+// Base class for a
+class BaseMemoryAllocator : public MemoryAllocator {
+ public:
+  void* Allocate(size_t /*size*/) override {
```

Comment:
Can you explain why we need the "BaseMemoryAllocator". Looks like "JemallocNodumpAllocator", "MockMemoryAllocator", and "MemkindKmemAllocator" can directly inherited from "MemoryAllocator"

Reply:
The BaseMemoryAllocator implements the methods in a failure mode.  By extending this class, Jemalloc et al can define the "real" methods only in the case where the functionality is compiled in.  In other words, if compiled with Jemalloc disabled, the "Base" implementation is used.  If compiled with Jemalloc enabled, the "real" implementation is used.  

Otherwise, the code gets funky to allow the Jemalloc et al classes to exist but be empty for the implementations in which they are not enabled at compile time.

Reply:
You may add "The BaseMemoryAllocator implements the methods in a failure mode. By extending this class, Jemalloc et al can define the "real" methods only in the case where the functionality is compiled in. In other words, if compiled with Jemalloc disabled, the "Base" implementation is used. If compiled with Jemalloc enabled, the "real" implementation is used" as the comment for `BaseMemoryAllocator` and indicate that all the allocator implementation should inherited from `BaseMemoryAllocator` instead of `MemoryAllocator`